DOCSP-51407: Auth reorganization (#120)
mongodb/docs-kotlin-sync: source/security/authentication/scram.txt

You can use ``SCRAM`` to authenticate to MongoDB Atlas, MongoDB
Enterprise Advanced, and MongoDB Community Edition.

.. tip:: SCRAM Mechanisms

   To learn more about the ``SCRAM`` family of authentication mechanisms, see
   `RFC 5802 <https://tools.ietf.org/html/rfc5802>`__ and
   :wikipedia:`Salted Challenge Response Authentication Mechanism <Salted_Challenge_Response_Authentication_Mechanism>`
   on Wikipedia.

   For more information about the MongoDB implementation of ``SCRAM``, see
   :manual:`SCRAM </core/security-scram>` in the {+mdb-server+} manual.

.. _kotlin-sync-auth-default:
.. _kotlin-sync-auth-scramsha256:

Specify SCRAM-SHA-256 Authentication
------------------------------------

``SCRAM-SHA-256``, as defined by `RFC 7677 <https://www.rfc-editor.org/rfc/rfc7677>`__, 
encrypts your username and password with the ``SHA-256`` algorithm to authenticate
your user. This is the default authentication mechanism.

The examples in this section show how to specify this default authentication
mechanism and use the following placeholder values:

- ``db_username``: Your MongoDB database username.
- ``db_password``: Your MongoDB database user's password.

> - ``hostname``: The network address of your MongoDB deployment, accessible by your client.

Reviewer: vale:

```suggestion
- ``hostname``: The network address of your MongoDB deployment, open to your client.
```

         :dedent:
         :start-after: start-scramsha256-cred-string
         :end-before: end-scramsha256-cred-string

   .. tab::
      :tabid: MongoCredential

      To specify the default authentication mechanism by using the
      ``MongoCredential`` class, use the
      `createScramSha256Credential() <{+core-api+}/MongoCredential.html#createScramSha256Credential(java.lang.String,java.lang.String,char[])>`__
      method as shown in the following example:

      .. literalinclude:: /includes/security/authentication.kt
         :language: kotlin
         :dedent:
         :start-after: start-scramsha256-mongo-cred
         :end-before: end-scramsha256-mongo-cred

.. _kotlin-sync-auth-scramsha1:

Specify SCRAM-SHA-1 Authentication
----------------------------------

``SCRAM-SHA-1``, as defined by `RFC 5802 <https://www.rfc-editor.org/rfc/rfc5802>`__, 
encrypts your username and password with the ``SHA-1`` algorithm to authenticate
your user.

The examples in this section show how to specify this authentication
mechanism and use the following placeholder values:

- ``db_username``: Your MongoDB database username.
- ``db_password``: Your MongoDB database user's password.

> - ``hostname``: The network address of your MongoDB deployment, accessible by your client.

Reviewer: vale:

```suggestion
- ``hostname``: The network address of your MongoDB deployment, open to your client.
```
